### PR TITLE
update minSdk to 24, cleanup code

### LINKED ISSUE #2607: Stopping support for Android 6.x
Tusky will stop supporting Android 6.x at the beginnig of 2023
Current usage of Android 6.x according to Google Play is 0,4% (250 devices).

### PR DESCRIPTION
**Pull request by connyduck**
closes #2607

### CODE REVIEW

**Review by Lakoja — APPROVED** [review]

**Comment by nikclayton** [comment]
Will merge after v22 is released.

File: app/src/main/java/com/keylesspalace/tusky/util/StatusParsingHelper.kt [hunk 1]
```diff
@@ -36,31 +35,3 @@ fun String.parseAsMastodonHtml(tagHandler: TagHandler? = null): Spanned {
...
-            builder!!.append(ASCII_HYPHEN)
-        } else if (replacing) {
-            builder!!.append(character)
-        }
-    }
-    return if (replacing) builder else content
-}
-
-private const val SOFT_HYPHEN = '\u00ad'
-private const val ASCII_HYPHEN = '-'
```
Comment (Tak) [inline app/src/main/java/com/keylesspalace/tusky/util/StatusParsingHelper.kt:66]: 🎉

**Review by Tak — APPROVED** [review]

**Comment by charlag** [comment]
Did we check the usage? do we strictly need to do this?

**Comment by connyduck** [comment]
> Did we check the usage? do we strictly need to do this?

Usage is in the issue. I think the simpler code and app size improvements are easily worth it.

**Comment by charlag** [comment]
Sorry that I've missed the usage, I am a bit surprised tbh but I agree, it is totally worth it if that's the case. Too bad we don't have F-Droid statistics but I don't think it would make a difference.

**Pull request closed without merging**